src/common/endianutil.h

```diff
@@ -17,9 +17,24 @@
 
 #pragma once
 
+#include <type_traits>
+
+#include "types.h"
+
 namespace reone {
 
+Endianess getEndianess();
+
+/* only permits fundamental integral types or extended integral types */
+template <typename T>
+std::enable_if_t<std::is_integral<T>::value || std::is_floating_point<T>::value> swapBytes(T &val) {
+	char *c = reinterpret_cast<char*>(&val);
+	for (size_t i=0; i*2 < sizeof(T); ++i) std::swap(c[i], c[sizeof(T) - i - 1]);
+}
+
 template <class T>
-void swapBytes(T &val);
+void fixEndianess(T& val,  bool isBigEndian) {
```

Comment:
I don't think it belongs here. The whole point of `fixEndianess` was "swap bytes if endianess of the STREAM differs from endianess of the system".

Reply:
Interesting, I think I misunderstood its point completely ...

Reply:
What about this?

```c++
template <class T>
void fixEndianess(T& val,  Endianess to);
```

Reply:
Given we're on x86 and want to read a native (LE) int from a BE stream, "to" would not be a correct name in this example. Also, "fix" is an ambiguous term. Maybe `swapBytesIfNotSystemEndianess` will do?